src/mlpack/core/data/string_encoding.hpp

```diff
@@ -0,0 +1,166 @@
+/**
+ * @file string_encoding.hpp
+ * @author Jeffin Sam
+ *
+ * Definition of string encoding functions.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_STRING_ENCODING_HPP
+#define MLPACK_CORE_DATA_STRING_ENCODING_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "mlpack/core/boost_backport/boost_backport_string_view.hpp"
+#include <mlpack/core/data/encoding_policies/dictionary_encoding.hpp>
+#include <mlpack/core/data/encoding_policies/policy_traits.hpp>
+#include <utility>
+
+using MapType =   std::unordered_map<boost::string_view, size_t,
```

Comment:
Tiny issue: there are some extra spaces after `=`.

Reply:
Sorry about that.